libraries/AP_HAL/utility/packetise.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+  support for sending UDP packets on MAVLink packet boundaries.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+
+/*
+  return the number of bytes to send for a packetised connection
+ */
+uint16_t mavlink_packetise(ByteBuffer &writebuf, uint16_t n)
+{
+    int16_t b = writebuf.peek(0);
+    if (b != MAVLINK_STX_MAVLINK1 && b != MAVLINK_STX) {
+        /*
+          we have a non-mavlink packet at the start of the
+          buffer. Look ahead for a MAVLink start byte, up to 256 bytes
+          ahead
+         */
+        uint16_t limit = n>256?256:n;
+        uint16_t i;
+        for (i=0; i<limit; i++) {
+            b = writebuf.peek(i);
+            if (b == MAVLINK_STX_MAVLINK1 || b == MAVLINK_STX) {
+                n = i;
+                break;
+            }
+        }
+        // if we didn't find a MAVLink marker then limit the send size to 256
+        if (i == limit) {
+            n = limit;
+        }
+        return n;
+    }
+
+    uint8_t min_length = (b == MAVLINK_STX_MAVLINK1)?8:12;
+    // this looks like a MAVLink packet - try to write on
+    // packet boundaries when possible
+    if (n < min_length) {
+        // we need to wait for more data to arrive
+        return 0;
+    }
+
+    // the length of the packet is the 2nd byte, and mavlink
+    // packets have a 6 byte header plus 2 byte checksum,
```

Comment:
The comment is wrong for MAVLink2, and probably belongs up next to the declaration of `min_length`

Reply:
fixed, thanks